sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/export/CollectionHandle.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.internal.export;
+
+import java.util.AbstractSet;
+import java.util.BitSet;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+import javax.annotation.Nullable;
+
+/**
+ * A handle for a collection-pipeline of metrics.
+ *
+ * <p>This class provides an efficient means of leasing and tracking exporters.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public final class CollectionHandle {
+  /** The index of this handle. */
+  private final int index;
+
+  private CollectionHandle(int index) {
+    this.index = index;
+  }
+
+  /** Construct a new (efficient) mutable set for tracking collection handles. */
+  public static Set<CollectionHandle> mutableSet() {
+    return new CollectionHandleSet();
+  }
+
+  /** Construct a new (mutable) set consistenting of the passed in collection handles. */
+  public static Set<CollectionHandle> of(CollectionHandle... handles) {
+    Set<CollectionHandle> result = mutableSet();
+    for (CollectionHandle handle : handles) {
+      result.add(handle);
+    }
+    return result;
+  }
+
+  /**
+   * Construct a new supplier of collection handles.
+   *
+   * <p>Handles returned by this supplier should not be used with unique handles produced by any
+   * other supplier.
+   */
+  public static Supplier<CollectionHandle> createSupplier() {
+    return new Supplier<CollectionHandle>() {
+      private final AtomicInteger nextIndex = new AtomicInteger(1);
+
+      @Override
+      public CollectionHandle get() {
+        return new CollectionHandle(nextIndex.getAndIncrement());
+      }
+    };
+  }
+
+  /** Creates a new collection handle using the next bit. */
+  public static CollectionHandle create() {
+    return new CollectionHandle(nextIndex.getAndIncrement());
+  }
+
+  @Override
+  public int hashCode() {
+    return index;
+  }
+
+  @Override
+  public boolean equals(@Nullable Object other) {
+    if (this == other) {
+      return true;
+    }
+    if (other == null) {
+      return false;
+    }
+    if (!(other instanceof CollectionHandle)) {
+      return false;
+    }
+    return index == ((CollectionHandle) other).index;
+  }
+
+  @Override
+  public String toString() {
+    return "CollectionHandle(" + index + ")";
+  }
+
+  // TODO: Better mechanism of this?
+  private static final AtomicInteger nextIndex = new AtomicInteger(1);
```

Comment:
move this up to the top of the class, please. :) Also, why not start at zero, and call `incrementAndGet()` when needed?

Reply:
Removed